src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -612,22 +612,23 @@ public ResultWrapper<IEnumerable<FilterLog>> eth_getFilterLogs(UInt256 filterId)
         CancellationTokenSource cancellationTokenSource = new(_rpcConfig.Timeout);
         CancellationToken cancellationToken = cancellationTokenSource.Token;
 
-        int id = (int)filterId;
-
         try
         {
-            if (_blockchainBridge.FilterExists(id))
+            int id;
+            if (filterId > int.MaxValue || !_blockchainBridge.FilterExists(id = (int)filterId))
```

Comment:
This `id = (int)filterId)` is very hidden and side-effecty, can we make it more visible?

Reply:
Moved one line up with a more explict check.